<launch>

  <arg name="dataroot" value="/data2/sgslam"/>
  <arg name="agentA_scene" value="uc0204_00a"/>
  <arg name="agentB_scene" value="uc0204_00b"/>
  <!-- <arg name="agentA_scene" value="ab0201_03a"/> -->
  <!-- <arg name="agentB_scene" value="ab0201_03c"/> -->

  <!-- x=11 , yaw=3.14 -->
  <arg name="agentA_agnetB_x" value="10.8"/>
  <arg name="agentA_agentB_y" value="0.9"/>
  <arg name="agentA_agentB_z" value="-4"/>
  <arg name="agentA_agentB_yaw" value="3.14"/>

  <group ns="viz">
    <param name="edge_width" value="0.03"/>
    <param name="edge_color/r" value="0.0"/>
    <param name="edge_color/g" value="1.0"/>
    <param name="edge_color/b" value="1.0"/>
    <param name="centroid_size" value="0.3"/>
  </group>

  <!-- inital tf to enable visulization -->
  <group ns="br">
    <param name="agentA/x" value="0.0"/>
    <param name="agentA/y" value="0.0"/>
    <param name="agentA/z" value="0.0"/>

    <param name="agentB/x" value="$(arg agentA_agnetB_x)"/>
    <param name="agentB/y" value="$(arg agentA_agentB_y)"/>
    <param name="agentB/z" value="$(arg agentA_agentB_z)"/>
    <param name="agentB/yaw" value="$(arg agentA_agentB_yaw)"/>
  </group>

  <node pkg="sgloop_ros" name="agentA" type="PoseGraphNode" output="screen" clear_params="true">
      <param name="src_name" value="agentA"/>
      <param name="ref_name" value="agentB"/>
      <param name="camera_marker/r" value="0.82"/>
      <param name="camera_marker/g" value="0.2"/>
      <param name="camera_marker/b" value="0.82"/>
      <param name="camera_marker/scale" value="0.2"/>
      <param name="camera_marker/line_width" value="0.05"/>
      <param name="pose_graph_folder" value="/data2/sfm/$(arg agentA_scene)-$(arg agentB_scene)/pose_graph"/>
      <param name="src_scene_dir" value="$(arg dataroot)/scans/$(arg agentA_scene)"/>
      <param name="gt_file" value="$(arg dataroot)/gt/$(arg agentA_scene)-$(arg agentB_scene).txt"/>
      <param name="frame_file_name" value="src_poses.txt"/>
      <param name="ate_threshold" value="3.0"/>

  </node>

  <node pkg="sgloop_ros" name="agentB" type="PoseGraphNode" output="screen" clear_params="true">
      <param name="src_name" value="agentB"/>
      <param name="ref_name" value="agentA"/>
      <param name="camera_marker/r" value="0.0"/>
      <param name="camera_marker/g" value="1.0"/>
      <param name="camera_marker/b" value="0.0"/>
      <param name="camera_marker/scale" value="0.2"/>
      <param name="camera_marker/line_width" value="0.05"/>
      <param name="frame_file_name" value="ref_poses.txt"/>

      <!-- <param name="map_only" value="true"/> -->
      <param name="pose_graph_folder" value="/data2/sfm/$(arg agentA_scene)-$(arg agentB_scene)/pose_graph"/>

      <param name="src_scene_dir" value="$(arg dataroot)/scans/$(arg agentB_scene)"/>

  </node>

  <node pkg="tf" type="static_transform_publisher" name="link_1st_agent" args="0 0 0 0 0 0 1 world agentA 100" />
  <node pkg="tf" type="static_transform_publisher" name="link_2nd_agent" args="$(arg agentA_agnetB_x) $(arg agentA_agentB_y) $(arg agentA_agentB_z) 0 0 1 0 world agentB 100" />

</launch>